gdc_client/download/client.py

```diff
@@ -19,10 +19,49 @@
 log = logging.getLogger('gdc-download')
 
 
-class GDCDownloadMixin(object):
+def fix_url(url):
+    # type: (str) -> str
+    """ Fix a url to be used in the rest of the program
+
+        example:
+            api.gdc.cancer.gov -> https://api.gdc.cancer.gov/
+    """
+    if not url.endswith('/'):
+        url = '{0}/'.format(url)
+
+    if not (url.startswith('https://') or url.startswith('http://')):
+        url = 'https://{0}'.format(url)
+
+    return url
+
+
+class GDCHTTPDownloadClient(Client):
```

Comment:
This is a lot more work than necessary. I started a cleanup project but got side-tracked. Here's how I started doing it with the mixin class and multiple inheritance. If you don't define an `__init__()` method then the superclass' method gets called.

https://github.com/NCI-GDC/gdc-client/blob/chore/code-style-cleanups/gdc_client/download/client.py

Reply:
Yea, I kinda noticed that but this is what actually pushed me to implement it like this https://github.com/NCI-GDC/gdc-client/blob/5f59d8502c4b4fe6c7a6709f4641bf980548bd91/gdc_client/download/client.py#L291

Which is intended to override the `parallel_download` method in the parcel `Client` class. This is an unusual way of doing inheritance, the `HTTPClient` also adds no new functionality, so can be removed and inherit directly from the `Client`. In a way we are basically on the same track, but keeping the `GDCUDTClient`